lib/Auth/Source/Twitter.php

```diff
@@ -82,92 +71,98 @@ public function __construct(array $info, array $config)
      * @param array &$state  Information about the current authentication.
      */
     public function authenticate(array &$state): void
+    {
+        $this->temporaryCredentials($state);
+    }
+
+
+    /**
+     * Retrieve temporary credentials
+     *
+     * @param array &$state  Information about the current authentication.
+     */
+    private function temporaryCredentials(array &$state): void
     {
         // We are going to need the authId in order to retrieve this authentication source later
         $state[self::AUTHID] = $this->authId;
 
-        $stateID = Auth\State::saveState($state, self::STAGE_INIT);
+        $stateId = base64_encode(Auth\State::saveState($state, self::STAGE_TEMP));
 
-        $consumer = new Consumer($this->key, $this->secret);
-        // Get the request token
-        $linkback = Module::getModuleURL('authtwitter/linkback.php', ['AuthState' => $stateID]);
-        $requestToken = $consumer->getRequestToken(
-            'https://api.twitter.com/oauth/request_token',
-            ['oauth_callback' => $linkback]
+        $server = new TwitterServer(
+            [
+                'identifier' => $this->key,
+                'secret' => $this->secret,
+                'callback_uri' => Module::getModuleURL('authtwitter/linkback.php')
+                    . '?AuthState=' . $stateId . '&force_login=' . strval($this->force_login),
```

Comment:
base64 encoding can include `=`, so this string must probably still be urlencoded?

Reply:
wrapping urlencode around the `$stateId` breaks things.. I've been struggling with this for days on end and this was the only way I could get it to work.
The thing is, this is not included in a URL, but in an HTTP-header and I read about different rules being applied there by the oAuth1 specs..